AEPAnalytics/Tests/AnalyticsTests.swift

```diff
@@ -13,22 +13,231 @@
 import XCTest
 import AEPServices
 @testable import AEPCore
-
 @testable import AEPAnalytics
 
 class AnalyticsAPITests: XCTestCase {
+    var mockRuntime: TestableExtensionRuntime!
+    var analytics: Analytics!
+
     override func setUp() {
-        EventHub.reset()
-        MockExtension.reset()
+        mockRuntime = TestableExtensionRuntime()
+        analytics = Analytics(runtime: mockRuntime)
+        analytics.onRegistered()
     }
 
-    private func registerMockExtension<T: Extension>(_ type: T.Type) {
-        let semaphore = DispatchSemaphore(value: 0)
-        EventHub.shared.registerExtension(type) { error in
-            XCTAssertNil(error)
-            semaphore.signal()
+    func testTrackState() {
+        // test
+        let trackData: [String: Any] = [
+            CoreConstants.Keys.STATE : "state",
+            CoreConstants.Keys.CONTEXT_DATA : [
+                "key1": "value1",
+                "key2": "value2"
+            ]
+        ]
+        let trackStateEvent = Event(name: "Generic track event", type: EventType.genericTrack, source: EventSource.requestContent, data: trackData)
+
+        mockRuntime.simulateSharedState(extensionName: AnalyticsConstants.Configuration.SHARED_STATE_NAME, event: trackStateEvent, data:
+            ([AnalyticsConstants.Configuration.GLOBAL_CONFIG_PRIVACY : PrivacyStatus.optedIn.rawValue], .set)
+        )
+        mockRuntime.simulateComingEvent(event: trackStateEvent)
+
+        XCTAssertEqual(mockRuntime.dispatchedEvents.count, 1)
+        let edgeEvent = mockRuntime.dispatchedEvents[0]
+        XCTAssertEqual(edgeEvent.type, EventType.edge)
+        XCTAssertEqual(edgeEvent.source, EventSource.requestContent)
+
+        let expectedData: [String:Any] = [
```

Comment:
would it be worth it to create a function that takes as an input a bool signaling a trackState, trackAction, or trackInternalAction and a string containing the tracked action or state? the function could then return the expectedData for each test.

Reply:
We just have one test for each API. I feel we can abstract the logic if we add more tests.

Reply:
sure sounds good.